File: rust_core/platform_support/ios/PlatformSupport/HWKeyStore/HWKeyStore.swift
//
//  HWKeyStore.swift
//  HWKeyStore
//
//  Created by Wallet Developer on 24/02/2023.
//

public final class HWKeyStore {
    public static let shared = HWKeyStore()

    private let keystore: KeyStore

    private init() {
        self.keystore = KeyStore()

        initHwKeystore(bridge: self.keystore)
    }
}

File: android/platform_support/src/main/java/com/example/platform_support/hwkeystore/HWKeyStore.kt
package com.example.platform_support.hwkeystore

import android.util.Log
import com.example.platform_support.hwkeystore.bridge.PlatformKeyStore
import uniffi.hw_keystore.KeyStoreBridge
import uniffi.hw_keystore.initHwKeystore

class HWKeyStore {
    companion object {
        val shared = HWKeyStore()

        private val keyStore: KeyStoreBridge = PlatformKeyStore()

        init {
            initHwKeystore(bridge = keyStore)

            //TESTING
            val key = keyStore.getOrCreateKey("first_key_id")
            Log.d(">>> HWKeyStore", ">>> SigningKey: $key")
        }
    }
}
